utils/zerovec/src/map/vecs.rs

```diff
@@ -13,19 +14,49 @@ use core::mem;
 
 /// Trait abstracting over [`ZeroVec`] and [`VarZeroVec`], for use in [`ZeroMap`](super::ZeroMap). You
 /// should not be implementing or calling this trait directly.
-pub trait ZeroVecLike<'a, T: ?Sized> {
+pub trait BorrowedZeroVecLike<'a, T: ?Sized> {
```

Comment:
Suggestion: you currently have these traits:

- `ZeroVecLike` has mutation methods specific to ZeroVec
- `BorrowedZeroVecLike` has methods that are common between both ZeroVec and ZeroVecBorrowed
- `BorrowedOnlyZeroVecLike` has read-only methods specific to ZeroVecBorrowed

Consider using the following naming scheme instead:

- `MutableZeroVecLike` for mutation methods
- `ZeroVecLike` for common methods
- `BorrowedZeroVecLike` for borrow-specific methods